### PR TITLE
[mnt] copy files to the wf directory (closes # 145)

### PR DESCRIPTION
**Pull request by djarecka**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

if files are in a workflow result, the files are copied to the workflow directory, using copyfile with `copy=File`, `use_hardlink=True`

## Checklist

- [x] All tests passing
- [x] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [x] My code follows the code style of this project 
(we are using `black`: you can `pip install pre-commit`, 
run `pre-commit install` in the `pydra` directory 
and `black` will be run automatically with each commit)

## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: pydra/engine/helpers.py [hunk 1]
```diff
@@ -112,13 +112,27 @@ def save(task_path: Path, result=None, task=None):
...
     if task:
         with (task_path / "_task.pklz").open("wb") as fp:
             cp.dump(task, fp)
 
 
+def copyfile_workflow(wf_path, result):
+    """ if file in the wf results, the file will be copied to the workflow directory"""
+    for field in attr_fields(result.output):
+        value = getattr(result.output, field.name)
+        if is_existing_file(value):
```
Comment (satra) [inline pydra/engine/helpers.py]: this doesn't take containers (lists, tuples, dicts) into account.

Comment (djarecka) [reply pydra/engine/helpers.py]: it doesn't

Comment (djarecka) [reply pydra/engine/helpers.py]: i could work on this tomorrow, didn't think about it

Comment (djarecka) [reply pydra/engine/helpers.py]: actually i'm not sure, let me  check

**Pull request merged**